calico_kubernetes/calico_kubernetes.py

```diff
@@ -726,15 +728,55 @@ def run_protected():
     """
     Runs the plugin, intercepting all exceptions.
     """
-    # Configure logging
-    configure_logger(logger, LOG_LEVEL, True)
-    configure_logger(pycalico_logger, LOG_LEVEL, False)
+    # Parse arguments and configure logging
+    global logger
+    mode = sys.argv[1]
+
+    # Append a stdout logging handler to log to the Kubelet.
+    # We cannot do this in the status hook because the Kubelet looks to
+    # stdout for status results.
+    if mode != 'status':
+        configure_stdout_logger(logger=logger,
+                                log_level=logging.INFO,
+                                root_logger=True)
+        configure_stdout_logger(logger=pycalico_logger,
+                                log_level=logging.INFO,
+                                root_logger=False)
+
+    # Adapt the logger to append the Docker ID to logs.
+    # Docker ID is not supplied in `init`, so do not include the it in logs.
+    if mode != 'init':
+        namespace = sys.argv[2].replace('/', '_')
+        pod_name = sys.argv[3].replace('/', '_')
+        docker_id = sys.argv[4]
+
+        configure_logger(logger=logger,
+                         log_level=LOG_LEVEL,
+                         log_format=DOCKER_ID_ROOT_LOG_FORMAT)
+        configure_logger(logger=pycalico_logger,
+                         log_level=LOG_LEVEL,
+                         log_format=DOCKER_ID_LOG_FORMAT)
+
+        logger = logging.LoggerAdapter(logger, {'docker_id': str(docker_id)[:12]})
+        logger = logging.LoggerAdapter(pycalico_logger, {'docker_id': str(docker_id)[:12]})
```

Comment:
Curses. Maybe we need to rethink this approach then. Instead of a LoggingAdapter, what about a Filter? It can cover similar use-cases. See https://docs.python.org/2/howto/logging-cookbook.html#using-filters-to-impart-contextual-information.

I think that might be clearer, because you can then do

```python
log_filter = IdentifierFilter(docker_id='xxx')
logger.addFilter(log_filter)
pycalico_logger.addFilter(log_filter)
```